ui/src/components/table/__tests__/QTable.spec.js

```diff
@@ -291,6 +434,138 @@ describe('Table API', () => {
+          )
+        })
+
+        it('should add row style (Array)', () => {
+          const style = 'font-size: 20px;'
+
+          mount(QTable, {
+            props: {
+              rows,
+              columns,
+              tableRowStyle: [ style ]
+            }
+          })
+
+          cy.get('.q-table tbody tr').each((tr) =>
+            cy.wrap(tr).should('have.attr', 'style', style)
+          )
+        })
+
+        it('should add row style (Object)', () => {
+          const style = 'font-size: 20px;'
+
+          mount(QTable, {
+            props: {
+              rows,
+              columns,
+              tableRowStyle: { fontSize: '20px' }
+            }
+          })
+
+          cy.get('.q-table tbody tr').each((tr) =>
+            cy.wrap(tr).should('have.attr', 'style', style)
+          )
+        })
+
+        it('should add row style (Function)', () => {
+          const style = 'font-size: 20px;'
+
+          mount(QTable, {
+            props: {
+              rows,
+              columns,
+              tableRowStyle: () => style
+            }
+          })
+
+          cy.get('.q-table tbody tr').each((tr) =>
+            cy.wrap(tr).should('have.attr', 'style', style)
+          )
+        })
+      })
+
+      describe('(prop): table-row-class', () => {
+        it('should add row classes (String)', () => {
+          const stringRowClasses = 'row-class1 row-class2'
+
+          mount(QTable, {
+            props: {
+              rows,
+              columns,
+              tableRowClass: stringRowClasses
+            }
+          })
+
+          cy.get('.q-table tbody tr').each((tr) =>
+            stringRowClasses.split(' ').forEach((cls) => cy.wrap(tr).should('have.class', cls))
+          )
+        })
+
+        it('should add row class (Array)', () => {
+          const arrayRowClasses = [ 'row-class1', 'row-class2' ]
+
+          mount(QTable, {
+            props: {
+              rows,
+              columns,
+              tableRowClass: arrayRowClasses
+            }
+          })
+
+          cy.get('.q-table tbody tr').each((tr) =>
+            arrayRowClasses.forEach((rowClass) => cy.wrap(tr).should('have.class', rowClass))
+          )
+        })
+
+        it('should add row class (Object)', () => {
+          const objectRowClasses = { 'row-class1': true, 'row-class2': true }
+
+          mount(QTable, {
+            props: {
+              rows,
+              columns,
+              tableRowClass: objectRowClasses
+            }
+          })
+
+          cy.get('.q-table tbody tr').each((tr) =>
+            Object.keys(objectRowClasses).forEach((rowClass) => cy.wrap(tr).should('have.class', rowClass))
+          )
+        })
+
+        it('should add row class (Function)', () => {
+          const rowClass = 'row-class'
+
+          mount(QTable, {
+            props: {
+              rows,
+              columns,
+              tableRowClass: () => rowClass
```

Comment:
This isn't testing the whole feature, you should use the row data to generate something for 2 different rows and then assert the desired outcome has been obtained for both

Reply:
Updated tests for both `rowStyle` and `rowClass`